### PR TITLE
CI for learn-layer5

### PR DESCRIPTION
**Pull request by kanishkarj**
Github CI to Build and push docker image.

### CODE REVIEW

**Comment by kanishkarj** [comment]
@leecalcote

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      with:
+        fetch-depth: 1
+    - name: Docker login
+      if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master') && success()
+      uses: azure/docker-login@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Docker build & push
+      if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master') && success()
```
Comment (leecalcote) [inline .github/workflows/ci.yml:26]: Nice, so we only push to Docker Hub when making a versioned release, yeah?

Comment (kanishkarj) [reply .github/workflows/ci.yml:26]: Actually, I don't mind either way. Should I make it push every time we merge to master?

Comment (kanishkarj) [reply .github/workflows/ci.yml:26]: Actually, I don't mind either way. Should I make it push every time we merge to master?

Comment (leecalcote) [reply .github/workflows/ci.yml:26]: @kanishkarj no, this type of control is great. @naturalett can you implement this same logic in Meshery repo?

Comment (leecalcote) [reply .github/workflows/ci.yml:26]: @rav121, fyi

Comment (naturalett) [reply .github/workflows/ci.yml:26]: @leecalcote Sure but I will wait until this PR will be merged. Just to be on the safe side.

**Pull request merged**